### PR TITLE
Fix: remove warnings coming from ss3 source code. Details below

### PR DESCRIPTION
**Pull request by k-doering-NOAA**
## What issue(s) does this PR address? Describe and add issue numbers, if applicable.
Issue was not created, but while working on #123, it became clear to me that some of the warning messages coming from SS3 could be taken care of. This will make it easier to identify new/different warnings in the future using github actions, and also cleans up the codebase a bit. In particular, the changes made are:

- add fallthrough statement to note implicit fallthrough is intentional
- get rid of maybe uninit warning
- use multiline comment instead of single line comment to get rid of warning
- remove parm_adjust_method from create_timevary because unused
- refactor: clean up comments for time varying so only in 1 place (no warning but found while getting rid of parm_adjust_method)
- try to fix warning re using wrong type in for statements
- use int instead of unsigned
- remove unused variable bio_t_base_from function

Link issue(s) here: n/a

## What tests have been done? Upload any model input files created for testing in a zip file, if possible.
periodically, pushed changes to github and checked that the change got rid of the warning using github actions run (note: squashed commits to simplify the PR).

## What tests/review still need to be done? Who can do it, and by when is it needed (ideally)?
@Rick-Methot-NOAA should check that none of the changes to fix the warnings are problematic.

Also, Check with the github action that the only remaining warnings are marked as [-Wdeprecated-copy] remain. 

I think some work that needs to be done before merging this in is revising the github action (probably in a branch in the workflow repository), so that it can do this same check automatically. The number of warnings expected will also be less than previously expected. I can work on this once Rick verifies the changes are not problematic.

## Check which is true. This PR requires:

- [x] no further changes to r4ss
- [x] no further changes to the manual
- [x] no further changes to SSI (the SS3 GUI)
- [ ] no further changes to the stock synthesis change log (new features, bug reports)

### CODE REVIEW

File: SS_selex.tpl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // SS_Label_file  #11. **SS_selex.tpl**
 // SS_Label_file  # * <u>get_selectivity()</u>  // does length and age selectivity and retention for all fleets
 // SS_Label_file  # * <u>make_fish_selex()</u>  // for all fleets in a particular season, does the dot product of length distribution with length selectivity and retention vectors to calculate equivalent mean quantities at age for each platoon
-
+
```
Comment (k-doering-NOAA) [inline SS_selex.tpl:4]: something weird is going on with the line endings. I will try to fix.

Comment (k-doering-NOAA) [reply SS_selex.tpl:4]: Github told me when I tried to edit inline that this file had a mix of line endings, which is why this is happening. It is changing them all to windows style (CRLF, which I think our other files are also using). So I think it is ok as is, as even editing this file on github would result in these changes.

**Review by Rick-Methot-NOAA — APPROVED** [review]
Changes looks good.
Stylistically, should block comments have the /* and */ on separate lines, or OK to have on same line as part of the comment?  I see both approaches being used.
I see some of the block comments getting deleted, which seems OK.  Those blocks probably were copies of other blocks in order to put the variable descriptions in close proximity to code under development.  So, OK to now delete all but one such instance.

**Comment by k-doering-NOAA** [comment]
Thanks @Rick-Methot-NOAA ! Glad you don't see anything that seems harmful. I will work on revising the github action warning run so it takes into account the fewer warnings.

> Stylistically, should block comments have the /* and */ on separate lines, or OK to have on same line as part of the comment? I see both approaches being used.

That's a good question to probably address when we decide on a style guide (#311). I don't have a strong preference either way, as someone who does not use block comments very often!

> I see some of the block comments getting deleted, which seems OK. Those blocks probably were copies of other blocks in order to put the variable descriptions in close proximity to code under development. So, OK to now delete all but one such instance.

Yep, that was intentional on my part. I thought it made the most sense to define the variables once by the function (as you would do with doxygen, for example), rather than have it copy/pasted many times on the function calls, as you pointed out.

**Review by Rick-Methot-NOAA — APPROVED** [review]
I approve these changes.

**Pull request merged**